cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -221,3 +226,108 @@ func selectNodePool(workerType string, nodePools []astro.NodePool, out io.Writer
 		return nodePoolID, errToReturn
 	}
 }
+
+// Delete deletes the specified worker queue from the deployment
+// user gets prompted if no deployment was specified
+// user gets prompted if no name for the queue to delete was specified
+// An errQueueDoesNotExist is returned if queue to delete does not exist
+// An errCannotDeleteDefaultQueue is returned if a user chooses the default queue
+func Delete(ws, deploymentID, deploymentName, name string, force bool, client astro.Client, out io.Writer) error {
+	var (
+		requestedDeployment astro.Deployment
+		err                 error
+		queueToDelete       *astro.WorkerQueue
+		listToDelete        []astro.WorkerQueue
+		queue               astro.WorkerQueue
+	)
+	// get or select the deployment
+	requestedDeployment, err = deployment.GetDeployment(ws, deploymentID, deploymentName, client)
+	if err != nil {
+		return err
+	}
+
+	// prompt for queue name if one was not provided
+	if name == "" {
+		name, err = selectQueue(requestedDeployment.WorkerQueues, out)
+		if err != nil {
+			return err
+		}
+	}
+	// check if default queue is being deleted
+	if name == "default" {
+		return errCannotDeleteDefaultQueue
+	}
+	queueToDelete = &astro.WorkerQueue{
+		Name:      name,
+		IsDefault: false, // cannot delete a default queue
+	}
+
+	if QueueExists(requestedDeployment.WorkerQueues, queueToDelete) {
+		if !force {
+			i, _ := input.Confirm(
+				fmt.Sprintf("\nAre you sure you want to delete the %s worker queue?", ansi.Bold(queueToDelete.Name)))
```

Comment:
```suggestion
				fmt.Sprintf("\nAre you sure you want to delete the %s worker queue? If there are any tasks in your DAGs assigned to this Worker Queue, the tasks might get stuck in a queued state and fail to execute.", ansi.Bold(queueToDelete.Name)))
```

Reply:
This is the same message the UI gives when you delete a worker queue. I think it would be good to warn the users of what could go wrong